## SmartTel Retention Intelligence: A Decision Intelligence Platform for Telecom Customer Retention

### Introduction

Customer retention has become one of the most important strategic priorities in the telecommunications industry. Acquiring new customers is significantly more expensive than retaining existing ones, making customer churn a direct threat to recurring revenue and long-term profitability. While predicting which customers are likely to leave is valuable, effective retention requires a deeper understanding of customer behavior, service usage patterns, and the factors that contribute to churn.

This project, SmartTel Retention Intelligence, aims to develop an end-to-end decision intelligence system that helps telecom providers identify customers at risk of churning, understand the drivers behind churn, prioritize high-value customers, and support data-driven retention strategies. Rather than treating churn prediction as a standalone machine learning problem, the project frames it as a business decision-making challenge where marketing budgets, customer value, and intervention effectiveness must all be considered.


The analysis uses the IBM Telco Customer Churn dataset, a widely used benchmark dataset containing demographic information, subscribed services, account details, billing information, and churn status for over 7,000 telecom customers. The target variable is Churn, indicating whether a customer discontinued the service.

#### Project Goal

To develop an end-to-end, explainable decision intelligence platform that enables telecom companies to proactively reduce customer churn by identifying at-risk customers, understanding the drivers of churn, prioritizing high-value retention opportunities, and estimating the financial impact of targeted retention strategies.

#### Project Objectives

- Reduce customer churn by identifying customers who are most likely to discontinue their services before they leave.
- Improve retention decision-making by providing actionable insights into the key factors influencing customer churn.
- Increase the effectiveness of retention campaigns by prioritizing customers who are both at high risk of churn and valuable to the business.
- Optimize the allocation of marketing resources by identifying where retention efforts are likely to generate the highest business returns.
- Minimize revenue loss associated with customer churn through early intervention and targeted retention strategies.
- Support strategic business planning by enabling stakeholders to simulate different retention scenarios and evaluate their financial impact before implementation.
-- 


#### Import Libraries

In [1]:
import pandas as pd
import numpy as np

import matplotlib.pyplot as plt
import seaborn as sns

import kagglehub
from kagglehub import KaggleDatasetAdapter

import warnings
warnings.filterwarnings('ignore')
import os


#### Load the Dataset

In [2]:
# Set the path to the file you'd like to load
file_path = "WA_Fn-UseC_-Telco-Customer-Churn.csv"

# Load the latest version
df = kagglehub.dataset_load(
  KaggleDatasetAdapter.PANDAS,
  "blastchar/telco-customer-churn",
  file_path,
  # Provide any additional arguments like 
  # sql_query or pandas_kwargs. See the 
  # documenation for more information:
  # https://github.com/Kaggle/kagglehub/blob/main/README.md#kaggledatasetadapterpandas
)

print("First 5 records:\n",  df.head())

    customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   
3  7795-CFOCW    Male              0      No         No      45           No   
4  9237-HQITU  Female              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ... DeviceProtection  \
0  No phone service             DSL             No  ...               No   
1                No             DSL            Yes  ...              Yes   
2                No             DSL            Yes  ...               No   
3  No phone service             DSL            Yes  ...              Yes   
4                No     Fiber optic             No  ...               No   

  TechSupport StreamingTV StreamingMovies        Contract Pap

In [3]:
df.shape

(7043, 21)

The dataset contains 7,043 customer records and 21 attributes describing customer demographics, subscribed services, billing information, and churn status.